test/jest/prometheus.spec.ts

```diff
@@ -26,18 +26,34 @@ describe("Prometheus and Alertmanager", () => {
   });
 
   test("alert manager should be firing watchdog alert", async () => {
-    // fetch active alerts with alertname="Watchdog"
-    const response = await fetch(
-      `${alertmanagerProxy.url}/api/v2/alerts/groups?filter=alertname%3D%22Watchdog%22&silenced=false&inhibited=false&active=true`,
-    );
-    expect(response.status).toBe(200);
+    // The Watchdog configuration uses group_interval: 5m and group_wait: 30s. This test might be executed too quickly
+    // to catch this. Therefore, we retry this test over the course of 6 mins (36 retries of 10s each). The absolute
+    // test timeout is set to 7 minutes to accommodate extra execution delays.
+    const maxRetries = 36;
+    const delay = 10000;
+    let success = false;
 
-    const body = (await response.json()) as {
-      alerts: Array<{ status: { state: string } }>;
-    }[];
-    expect(body[0]).toBeDefined();
-    expect(body[0].alerts[0].status.state).toEqual("active");
-  });
+    for (let retries = 0; retries < maxRetries; retries++) {
+      const response = await fetch(
+        `${alertmanagerProxy.url}/api/v2/alerts/groups?filter=alertname%3D%22Watchdog%22&silenced=false&inhibited=false&active=true`
+      );
+
+      if (response.status === 200) {
+        const body = (await response.json()) as {
+          alerts: Array<{ status: { state: string } }>;
+        }[];
+
+        if (body[0]?.alerts[0]?.status.state === "active") {
+          success = true;
+          break;
+        }
+      }
```

Comment:
Is there any specific status we might be able to check to differentiate between "it's broken" versus "we're waiting for it to fire"? I suspect probably not but figured I'd ask - if it's actually broken it would be nice to fail execution early rather than waiting 7 mins.